splink/comparison_helpers.py

```diff
@@ -0,0 +1,425 @@
+import re
+
+import altair as alt
+import duckdb
+import pandas as pd
+import phonetics
+
+
+def comparator_score(str1, str2, decimal_places=2):
+    """Helper function to give the similarity between two strings for
+    the string comparators in splink.
+
+    Examples:
+        ```py
+        import splink.comparison_helpers as ch
+
+        ch.comparator_score("Richard", "iRchard")
+        ```
+    """
+    con = duckdb.connect()
+
+    sql = f"""
+        select
+        '{str1}' as string1,
+        '{str2}' as string2,
+        levenshtein('{str1}', '{str2}') as levenshtein_distance,
+        damerau_levenshtein('{str1}', '{str2}') as damerau_levenshtein_distance,
+        ROUND(jaro_similarity('{str1}', '{str2}'), {decimal_places}) as jaro_similarity,
+        ROUND(jaro_winkler_similarity(
+            '{str1}', '{str2}'), {decimal_places}
+            ) as jaro_winkler_similarity,
+        ROUND(jaccard('{str1}', '{str2}'), {decimal_places}) as jaccard_similarity
+    """
+    return con.execute(sql).fetch_df()
+
+
+def distance_match(distance, threshold):
+    if distance <= threshold:
+        return True
+    else:
+        return False
+
+
+def similarity_match(similarity, threshold):
+    if similarity >= threshold:
+        return True
+    else:
+        return False
+
+
+def threshold_match(comparator, score, distance_threshold, similarity_threshold):
+    if re.search("distance", comparator):
+        return distance_match(score, distance_threshold)
+    elif re.search("similarity", comparator):
+        return similarity_match(score, similarity_threshold)
+
+
+def comparator_score_df(list, col1, col2, decimal_places=2):
+    """Helper function returning a dataframe showing the string similarity
+    scores and string distances for a list of strings.
+
+    Examples:
+        ```py
+        import splink.comparison_helpers as ch
+
+        list = {
+                "string1": ["Stephen", "Stephen","Stephen"],
```

Comment:
overly pedantic, but

```suggestion
                "string1": ["Stephen", "Stephen", "Stephen"],
```